appendable-rs/appendable/Cargo.toml

```diff
@@ -6,3 +6,10 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+protocol = { path="../protocol" }
+serde_json = "1.0.111"
```

Comment:
This isn't necessary anymore, right?

Reply:
So I'm still debating whether or not to include `serde_json`. My current work uses it to parse the `json` object and it allows me to cleanly pattern match like so:

```rust
let json: Value = serde_json::from_str(json_line).map_err(|e| e.to_string())?;

    let mut bytes_processed = 0;

    match json {
        Value::Object(obj) => {
            for (k, v) in obj {
                let field_offset = data_offset + (k.len() as u64) + 1; // account for : sym

                path.push(k.clone());
                let name = path.join(".");

                match v {
                    Value::Object(_) => {
                        let nested_json = serde_json::to_string(&v).map_err(|e| e.to_string())?;

                        let nested_bytes = handle_json_object(
                            &nested_json,
                            index_file,
                            path,
                            data_index,
                            field_offset,
                        )?;

                        bytes_processed += nested_bytes
                    }
                    Value::String(s) => {}
                    Value::Number(n) => {}
                    Value::Bool(b) => {}
                    Value::Null => {}
                    _ => return Err(format!("Unexpected type: {}", v)),
                }
            }
        }
        _ => {
            return Err("Expected JSON object".to_string());
        }
    }
```

But as you can see we might not be accurate with our bytes calculation. So I wanted to experiment with the `serde_json` dependency, and if it doesn't mix well, I'd attempt at writing a custom tokenizer

Reply:
Up to you, but I suspect since we need byte-precise parsing, it might be simpler to write a custom tokenizer. JSON is not too complicated. I'm surprised Rust doesn't have one in the standard library, actually.